src/librustc_typeck/check/_match.rs

```diff
@@ -867,6 +868,15 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             self.check_pat_walk(&field.pat, field_ty, def_bm, true);
         }
 
+        if tcx.sess.features.borrow().non_exhaustive {
+            // Require `..` if structure or enum has non_exhaustive attribute.
+            if adt.non_exhaustive && !adt.did.is_local() && !etc {
+                tcx.sess.span_err(
```

Comment:
I think this error should have its own error number.

Reply:
I've added error numbers in a WIP commit that I couldn't get compiling locally - messages regarding the diagnostic code not being registered despite my not being able to spot a difference in how they are being added. Would appreciate any thoughts here.